File: lib/domain/cultivation_plan/interactors/entry_schedule/crop_stage_snapshot.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Interactors
      module EntrySchedule
        CropStageSnapshot = Struct.new(
          :id,
          :name,
          :order,
          :temperature_requirement,
          keyword_init: true
        )
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/interactors/entry_schedule/crop_stage_snapshot.rs
//! Ruby: `Domain::CultivationPlan::Interactors::EntrySchedule::CropStageSnapshot`

use super::temperature_requirement_snapshot::TemperatureRequirementSnapshot;

#[derive(Debug, Clone, PartialEq)]
pub struct CropStageSnapshot {
    pub id: i64,
    pub name: String,
    pub order: i32,
    pub temperature_requirement: Option<TemperatureRequirementSnapshot>,
}
